src/custom-elements/src/components/link/link.tsx

```diff
@@ -0,0 +1,102 @@
+import { Component, Element, Prop, State, Watch, h } from '@stencil/core';
+
+@Component({
+  tag: 'chi-link',
+  styleUrl: 'link.scss',
+  scoped: true
+})
+export class Link {
+  @State() slotLinkContent = true;
+
+  @Element() el: HTMLElement;
+
+  /**
+   *  to set a links destination url.
+   */
+  @Prop({ reflect: true }) href: string = '#';
+
+  /**
+   *  to set the language of a links destination url.
+   */
+  @Prop({ reflect: true }) hreflang: string;
+
+  /**
+   *  to provide alternative text in case of icon links.
+   */
+  @Prop({ reflect: true }) alternativeText: string;
+
+  /**
+   *  to render a link as a prominent call to action.
+   */
+  @Prop({ reflect: true }) cta = false;
+
+  /**
+   *  to disable a link.
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  /**
+   *  to prompt users to save a page or document instead of navigating to it.
+   */
+  @Prop({ reflect: true }) download: string;
+
+  /**
+   *  to set the relationship between the current page and the linked page.
+   */
+  @Prop({ reflect: true }) rel: string;
+
+  /**
+   *  to set size of a link { xs, sm, md, lg, xl }.
+   */
+  @Prop({ reflect: true }) size: string;
+
+  /**
+   *  to set target of a link { self, blank, parent, top }.
+   */
+  @Prop({ reflect: true }) target: string;
+
+  @Watch('size')
+  sizeValidation(newValue: string) {
+    if (newValue && !['', 'xs', 'sm', 'md', 'lg', 'xl'].includes(newValue)) {
+      throw new Error(`${newValue} is not a valid size for link. Valid values are xs, sm, md, lg, xl, ''. `);
```

Comment:
I think including an empty string in the comparison is not necessary, since _newValue_ would return false. Because of that, I would reword the Error like this:
```suggestion
    if (!!newValue && !['', 'xs', 'sm', 'md', 'lg', 'xl'].includes(newValue)) {
      throw new Error(`${newValue} is not a valid size for link. If provided, valid values are xs, sm, md, lg, or xl. `);
```